dev-go/act/act-1.6.0.ebuild

```diff
@@ -0,0 +1,82 @@
+# Copyright 1999-2021 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+inherit golang-base toolchain-funcs
+
+DESCRIPTION="Automated Component Toolkit code generator"
+HOMEPAGE="https://github.com/Autodesk/AutomaticComponentToolkit"
+SRC_URI="https://github.com/Autodesk/AutomaticComponentToolkit/archive/v${PV}.tar.gz -> ${P}.tar.gz"
+S="${WORKDIR}/AutomaticComponentToolkit-${PV}"
+
+LICENSE="BSD-2"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+
+RESTRICT="strip test"
+
+DOCS=( CODE_OF_CONDUCT.md CONTRIBUTING.md README.md )
+
+# Package does not follow the usual go directory structure
+# Functions borrowed from dev-lang/go ebuild.
+go_arch() {
+	# By chance most portage arch names match Go
+	local portage_arch=$(tc-arch $@)
+	case "${portage_arch}" in
+		x86)	echo 386;;
+		x64-*)	echo amd64;;
+		ppc64) [[ $(tc-endian $@) = big ]] && echo ppc64 || echo ppc64le ;;
+		s390)	echo s390x ;;
+		*)		echo "${portage_arch}";;
+	esac
+}
+
+go_arm() {
+	case "${1:-${CHOST}}" in
+		armv5*) echo 5;;
+		armv6*) echo 6;;
+		armv7*) echo 7;;
+		*)
+			die "unknown GOARM for ${1:-${CHOST}}"
+			;;
+	esac
+}
+
+go_os() {
+	case "${1:-${CHOST}}" in
+		*-linux*)	echo linux;;
+		*-darwin*)	echo darwin;;
+		*-freebsd*)	echo freebsd;;
+		*-netbsd*)	echo netbsd;;
+		*-openbsd*)	echo openbsd;;
+		*-solaris*)	echo solaris;;
+		*-cygwin*|*-interix*|*-winnt*)
+				echo windows
+				;;
+		*)
+				die "unknown GOOS for ${1:-${CHOST}}"
+				;;
+	esac
+}
+
+go-tuple() {
+	echo "$(go_os $@)_$(go_arch $@)"
+}
+
+src_compile() {
+	export GOARCH=$(go_arch)
+	export GOOS=$(go_os)
+	if [[ ${ARCH} == arm ]]; then
+		export GOARM=$(go_arm)
+	fi
+
+	cd "${S}"/Source
```

Comment:
`|| die`. You can use `pushd -> popd` if you like it better.

Reply:
Good catch. Forgot about `die` here